cmd/cloudFoundryCreateSpace_test.go

```diff
@@ -0,0 +1,74 @@
+package cmd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/cloudfoundry"
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCloudFoundryCreateSpace(t *testing.T) {
+	m := &mock.ExecMockRunner{}
+	s := mock.ShellMockRunner{}
+
+	cf := cloudfoundry.CFUtils{Exec: m}
+	cfUtilsMock := cloudfoundry.CfUtilsMock{}
+	var telemetryData telemetry.CustomData
+
+	config := cloudFoundryCreateSpaceOptions{
+		CfAPIEndpoint: "https://api.endpoint.com",
+		CfOrg:         "testOrg",
+		CfSpace:       "testSpace",
+		Username:      "testUser",
+		Password:      "testPassword",
+	}
+
+	t.Run("CF login: Success", func(t *testing.T) {
+
+		err := runCloudFoundryCreateSpace(&config, &telemetryData, cf, &s)
+		if assert.NoError(t, err) {
+			assert.Contains(t, s.Calls[0], "yes '' | cf login -a https://api.endpoint.com -u testUser -p testPassword")
+		}
+	})
+
+	t.Run("CF login: failure case", func(t *testing.T) {
+
+		errorMessage := "cf login failed"
+
+		defer func() {
+			s.Calls = nil
+			s.ShouldFailOnCommand = nil
```

Comment:
For `CfUtilsMock` there is a [`Cleanup` method](https://github.com/SAP/jenkins-library/blob/7ea5b095551550ee89cf9901cda293bfa6802daa/pkg/cloudfoundry/Authentication.go#L132). Maybe it would make sense to create a method for `ShellMockRunner` too